### ISSUE TITLE
Accessibility – Improvement: Encryption Combobox not working with screen reader in the add backup  option using  new WebUI “ngclient”

### ISSUE THREAD

**Issue by citygabor** [duplicati/ngclient#100]
- [x] I have searched open and closed issues for duplicates.
- [x] I have searched the [forum](https://forum.duplicati.com/) for related topics.

----------------------------------------

## Environment info

Local
- **Duplicati version**: ServerVersionName : - 2.1.0.107_canary_2025-01-17

- **Operating system**: TrueNAS 24.10.1
- **Backend**: Debian

## Description

Hello Duplicati Team,

I would like to report an accessibility issue about the new “ngclient” WebUI version.
Once I am on the add backup page I cannot change the combobox content of the Encryption field on the new “ngclient” WebUI page.
ServerVersionName : - 2.1.0.107_canary_2025-01-17
OS: Truenas   2024.10.0

In the current legacy WebUI page I do not feel any difficult to change the Encryption type using screen reader and using my keyboard only.

What I could discover  in the legacy page the Encryption field is a combobox , but in the new “ngclient” page that field is act as an editable text field.
When I navigate to the Encryption field and hitting the enter key, nothing happens. As Every blind people only use keyboard without mouse and screen reader it is a key to open that combobox and select the right Encryption type for the backup.

You can open the new “ngclient” page and add the new backup and turning on your screen reader (Windows: Narrator  - Buit built in screen reader, NVDA, J.A.W.S., Linux: Orca, MacOS: VoiceOver.
Once the virtual keyboard is on the page and hit the “C” keyboard in this case the navigation cannot be made on the “ngclient”, but within the “legacy” page it can be navigate using the “C” key. In the new WebUI we can use the “E” key to navigate easily, but when I open the edit text box my screen reader is silence.

If you could test it on your own and find the way to make it working on the “add Backup” option, please fix it on the other menus/ options as well, where this edit text box is used.

Please contact me if you need any further details.

I could suggest to use the Accessibility Insight for Web extension to discover accessibility issues..
https://accessibilityinsights.io/

Many Thanks,
Regards,
Gabor

## Steps to reproduce

You can open the new “ngclient” page and add the new backup and turning on your screen reader (Windows: Narrator  - Buit built in screen reader, NVDA, J.A.W.S., Linux: Orca, MacOS: VoiceOver.
Once the virtual keyboard is on the page and hit the “C” keyboard in this case the navigation cannot be made on the “ngclient”, but within the “legacy” page it can be navigate using the “C” key. In the new WebUI we can use the “E” key to navigate easily, but when I open the edit text box my screen reader is silence.

In this case on the “ngclient” WebUI page the screen reader does not read the values of the Encryption type editable text field.

When I open the combobox with my screen reader using my keyboard only my screen reader should read every value of the encryption when I moving my cursor up or down.
- **Actual result**:
In this case on the “ngclient” WebUI page the screen reader does not read the values of the Encryption type editable text field.

- **Expected result**:
When I open the combobox with my screen reader using my keyboard only my screen reader should read every value of the encryption when I moving my cursor up or down.

**Comment by sp90** [comment]
This has been fixed

https://github.com/duplicati/ngclient/commit/6062f4eeae050fbadcc727128518cd4d5e6a761e

**Comment by citygabor** [comment]
Hello Team,

This is not fixed yet even in this version.
ServerVersion : 2.1.0.120

I have just updated recently and wanted to create a new test backup and the encryption combobox/Edit field acting as before.

When my screen user is enabled with virtual keyboard is on any browser and navigating into the Encryption edit field and hitting the enter key or the space bar I cannot changeing the encryption type to None or ASH, etc and my screen reader does not even read the available options.

Can you please check it again?

Please see what I could find with the Accessibilityinsight for web extension.

Title: WCAG 1.3.1: Ensure <li> elements are used semantically (li:nth-child(2))
Tags: Accessibility, WCAG 1.3.1, listitem

Issue: Ensure <li> elements are used semantically (listitem - https://accessibilityinsights.io/info-examples/web/listitem)

Target application: Duplicati ngclient - http://192.168.1.100/ngclient/backup/new/general

Element path: li:nth-child(2)

Snippet: <li class="option"> AES-256 encryption, built in </li>

How to fix: 
Fix any of the following:
  List item does not have a <ul>, <ol> parent element

Environment: Microsoft Edge version 137.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.46.0 (axe-core 4.10.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

Please test it your own as well to turn on the screen reader on your machine to see the behiviour.



I am not sure it helps you  to find the root cause but it worth to mentioned.

Many Thanks  for your help,
Regards,
Gabor
